neqo-transport/src/path.rs

```diff
@@ -389,7 +348,15 @@ impl Paths {
 
         self.paths.retain(|p| {
             let current = p.borrow().remote_cid.as_ref().unwrap().sequence_number();
-            if current < retire_prior {
+            if current < retire_prior
+                && p.borrow()
+                    .remote_cid
+                    .as_ref()
+                    .unwrap()
+                    .connection_id()
+                    .len()
```

Comment:
Are the two calls to `p.borrow()` necessary? Or would something along the lines of the below work as well?

``` rust
let current_remote_cid = p.borrow().remote_cid.as_ref.unwrap();
if current_remote_cid .sequence_number() < retire_prior && current_remote_cid.connection_id().len() != 0 {
```